lib/Dialect/SPIRV/Serialization/Serializer.cpp

```diff
@@ -618,6 +636,29 @@ Serializer::prepareBasicType(Location loc, Type type, uint32_t resultID,
     return success();
   }
 
+  if (auto structType = type.dyn_cast<spirv::StructType>()) {
+    for (uint32_t elementNum = 0; elementNum < structType.getNumElements();
+         ++elementNum) {
+      uint32_t elementTypeID = 0;
+      if (failed(processType(loc, structType.getElementType(elementNum),
+                             elementTypeID))) {
+        return failure();
+      }
+      operands.push_back(elementTypeID);
+      if (structType.hasLayout()) {
+        // Decorate each struct member with an offset
+        if (failed(processMemberDecoration(
+                resultID, elementNum, spirv::Decoration::Offset,
+                static_cast<uint32_t>(structType.getOffset(elementNum))))) {
+          return emitError(loc, "cannot decorate member: ")
```

Comment:
"cannot decorate " << elementNum << "-th member of " << structType << " with its offset" ?

Reply:
Thanks!